superset/viz.py

```diff
@@ -1653,6 +1653,7 @@ def run_extra_queries(self) -> None:
                 query_obj["orderby"] = [(metric, asc)]
             self.get_query_context_factory().create(
                 datasource={"id": self.datasource.id, "type": self.datasource.type},
+                form_data=self.form_data,
```

Comment:
This fixes https://github.com/apache/superset/issues/25006, form data wasn't being passed through to `raise_for_access`